File: Data Structures & Algorithms/daily-temperatures/submission-1.java
class Solution {
    public int[] dailyTemperatures(int[] temperatures) {
        Stack<int[]> stack = new Stack<>();
        int n = temperatures.length;
        int[] result = new int[n];

        for(int i=n-1;i>=0;i--){
            while(!stack.empty() && temperatures[i]>=stack.peek()[0])stack.pop();
            if(stack.empty()){
                result[i]=0;
            }
            else{
                result[i] = stack.peek()[1]-i;
            }
            stack.push(new int[]{temperatures[i],i});
        }
        return result;
    }
}

File: Data Structures & Algorithms/baseball-game/submission-1.cpp
class Solution {
public:
    int calPoints(vector<string>& operations) {
        stack<int> s;
        for(const string& operation: operations){
            if(operation=="+"){
                int first = s.top();
                s.pop();
                int second = s.top();
                s.push(first);
                s.push(first+second);
            }
            else if(operation=="D"){
                int first = s.top();
                s.push(2*first);
            }
            else if(operation=="C"){
                s.pop();
            }
            else{
                int no = stoi(operation);
                s.push(no);
            }
        }
        int ans = 0;
        while(!s.empty()){
            ans += s.top();
            s.pop();
        }
        return ans;
    }
};